gandi/cli/commands/config.py

```diff
@@ -0,0 +1,67 @@
+""" Configuration namespace commands. """
+
+import sys
+import os
+import subprocess
+
+import click
+
+from gandi.cli.core.cli import cli
+from gandi.cli.core.params import pass_gandi
+
+
+@cli.command()
+@click.option('-g', help='Get from global configuration only'
+                         '(default=env -> local -> global).',
+              is_flag=True, default=False)
+@click.argument('key')
+@pass_gandi
+def get(gandi, g, key):
+    """Display value of a given config key."""
+    val = gandi.get(key=key, global_=g)
+    if not val:
+        gandi.echo("No value found.")
+        sys.exit(1)
+    gandi.echo(val)
+
+
+@cli.command()
+@click.option('-g', help='Edit global configuration (default=local).',
+              is_flag=True, default=False)
+@click.argument('key')
+@click.argument('value')
+@pass_gandi
+def set(gandi, g, key, value):
+    """Update or create config key/value."""
+    gandi.configure(global_=g, key=key, val=value)
+
+
+@cli.command()
+@click.option('-g', help='Edit global configuration (default=local).',
+              is_flag=True, default=False)
+@pass_gandi
+def edit(gandi, g):
+    """Edit config file with prefered text editor"""
+    config_file = gandi.home_config if g else gandi.local_config
+    path = os.path.expanduser(config_file)
+    editor = gandi.get('editor', '/usr/bin/vi')
```

Comment:
Hardcoding the path of vi is wrong, some OS (used to?) have vi in a different path, like old redhat had vi in /bin don't know if they still do. In C you have a macro for that _PATH_VI defined in paths.h which of course is not available in python so I would go for a plain 'vi' and rely on the path to find the proper path.
Defaulting on 'vim' won't imho be a good idea, has you have the insurance that vi is installed on anything trying to respect POSIX (part of POSIX.1-2008) (ok I know no one care about posix :) but hey)
In anycase +1 on adding support for the EDITOR variable